base/sysinfo.jl

```diff
@@ -513,6 +512,19 @@ See documentation in [Handling Operating System Variation](@ref).
 """
 iswindows(os::Symbol) = (os === :Windows || os === :NT)
 
+"""
+    Sys.isWSL([os])
+
+Predicate for testing if the OS is the Windows subsystem for Linux (version 2 or the original),
+running under a derivative of Microsoft Windows NT.
+See documentation in [Handling Operating System Variation](@ref).
+
+!!! compat "Julia 1.12"
+    This function requires at least Julia 1.12.
+"""
+# https://superuser.com/questions/1749781/how-can-i-check-if-the-environment-is-wsl-from-a-shell-script
+isWSL(os::Symbol) = ispath("/proc/sys/fs/binfmt_misc/WSLInterop") # WSL sigil
```

Comment:
I think putting a comment there might break the docstring lookup

Reply:
Also, you aren't using the input argument, besides not defining the method you try to use elsewhere.

Reply:
Yes like this seems better:
https://github.com/JuliaLang/julia/issues/36354#issuecomment-647629429
>How about Sys.detectwsl?

I'll just abandon this/make a draft, since there's this other PR. If I don't see it move then I may work on it here later. There one implementation here outside of Julia:

https://github.com/tpapp/DefaultApplication.jl/pull/17/files